### PR TITLE
report errno on NanoSocket.send failure

### PR DESCRIPTION
**Pull request by John-Colvin**
should help investigating problems.

also, removed some imports that weren't being used.

### CODE REVIEW

File: source/nanomsg/wrap.d [hunk 1]
```diff
@@ -239,6 +235,7 @@ struct NanoSocket {
 
         const sent = () @trusted { return nn_send(_nanoSock, data.ptr, data.length, flags(blocking)); }();
         if(blocking) {
+            enforceNanoMsgRet(sent, file, line);
```
Comment (atilaneves) [inline source/nanomsg/wrap.d]: This is already covered by the exception being thrown below.

Comment (John-Colvin) [reply source/nanomsg/wrap.d]: this one checks for `>0` and reads errno (i.e. checking for nanomsg reporting failure), the next one checks for the exact length (i.e. nanomsg thinks it's fine but we're not happy with what it did)

Comment (atilaneves) [reply source/nanomsg/wrap.d]: `>0` and same as the length is the same check except for when the length is 0, which is not something anybody should be doing anyway. And if they are, they should expect to send 0 bytes. Checking errno is a good idea, but then I'd change it to `enforceNanoMsgRet(sent == data.length, file, line)`.

**Comment by John-Colvin** [comment]
I changed it to not check for the length matching at all.

`int nn_send (int s, const void *buf, size_t len, int flags)` always returns either `-1` or `len` (unless passed a special `buf` created by `nn_allocmsg` with the constant `NN_MSG` passed for `len`, which we definitely aren't here), so there's no information there. `enforceNanoMsgRet` checks for `<0` to signify failure, so we're good.

**Pull request merged**